pkg/azuredisk/azuredisk_v2.go

```diff
@@ -181,6 +187,14 @@ func (d *DriverV2) Run(endpoint, kubeConfigPath string, testBool bool) {
 	})
 
 	ctx, cancel := context.WithCancel(context.Background())
+	c := make(chan os.Signal, 2)
+	signal.Notify(c, []os.Signal{os.Interrupt, syscall.SIGTERM}...)
+	go func() {
```

Comment:
https://gobyexample.com/select

Reply:
select was not added because there is no need to identify what kind of signal is fed into the channel because it is going to be either an Interrupt or SIGTERM. As long as it's one of them, context is cancelled.

Reply:
Oh, I see that you don't want to differentiate between the signals but with this implementation unless you always expect to receive 2 signals, you are going to leak this goroutine. Is that the case that you are guaranteed to receive 2 signals?

Reply:
We expect this goroutine to run for the lifetime of the azurediskplugin process so leak wouldn't be a concern here. Second signal is like force-shutting the process while the first signal allows azurediskplugin to gracefully exit by cancelling context.

Reply:
It would be good to add a comment explaining this in the code.

In the course of shutdown by kubelet, will we receive the second signal? Is this behavior documented somewhere?

Reply:
https://cloud.google.com/blog/products/containers-kubernetes/kubernetes-best-practices-terminating-with-grace

As a matter of fact, it seems like Kubernetes sends a SIGKILL signal if the pod does not gracefully exit with a SIGTERM signal within the set `terminationGracePeriodsSeconds`. That said, waiting for the signal would be unnecessary and just cancelling the context upon receival of the first signal would be sufficient.

Reply:
Ah, thanks for the reference! Yes, waiting for the second signal here would mean that we would always be taking at least `terminateGracePeriodsSeconds` to terminate which is not ideal. We should just wait for the first signal as you propose.